src/libraries/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.cs

```diff
@@ -332,7 +341,13 @@ public void Ctor_SafeHandle_BasicPropertiesPropagate_Success(AddressFamily addre
 
             Assert.Equal(addressFamily, copy.AddressFamily);
             Assert.Equal(socketType, copy.SocketType);
-            Assert.Equal(protocolType, copy.ProtocolType);
+            ProtocolType expectedProtocolType = protocolType;
+            if (isRawPacket)
+            {
+                // raw packet doesn't support getting the protocol using getsockopt SO_PROTOCOL.
+                expectedProtocolType = ProtocolType.Unspecified;
```

Comment:
Does that mean Socket.ProtocolType throws in that case?  Should we hardcode it to return Unspecified, or is it ok failing?

Reply:
It doesn't throw, `getsockopt SO_PROTOCOL` returns success with a value of `AF_UNSPEC`.

Reply:
Ah, ok.